modules/apps/27-interchain-accounts/types/metadata.go

```diff
@@ -9,3 +16,59 @@ func NewMetadata(version, controllerConnectionID, hostConnectionID, accAddress s
 		Address:                accAddress,
 	}
 }
+
+// ValidateControllerMetadata performs validation of the provided ICS27 controller metadata parameters
+func ValidateControllerMetadata(ctx sdk.Context, channelKeeper ChannelKeeper, connectionHops []string, metadata Metadata) error {
```

Comment:
Originally I had a single `ValidateMetadata` function but depending on whether your in `OnChanOpenInit` or `OnChanOpenTry` the `connectionHops` will obviously correspond to opposing connection ends, and it would be comparing against the incorrect `metadata` field for controller/host. 

This can be consolidated to a single `ValidateMetadata` function when the connections IDs are removed from  `metadata`

Reply:
we could make a private function which takes in the controller connection ID and the host connection ID. ie just making the `ValidateControllerMetdata` get the connection and call the private function. Ditto for host. Then you deduplicate code

Reply:
Great suggestion. Updated!